src/biogeophys/BalanceCheckMod.F90

```diff
@@ -398,77 +568,75 @@ subroutine BalanceCheck( bounds, &
           end if
        end do
 
-       ! Water balance check
+       ! Water balance check at the column level
 
        do c = bounds%begc, bounds%endc
 
           ! add qflx_drain_perched and qflx_flood
           if (col%active(c)) then
 
-             errh2o(c) = endwb(c) - begwb(c) &
+             errh2o_col(c) = endwb_col(c) - begwb_col(c) &
                   - (forc_rain_col(c)        &
                   + forc_snow_col(c)         &
-                  + qflx_floodc(c)           &
-                  + qflx_sfc_irrig(c)        &
-                  + qflx_glcice_dyn_water_flux(c) &
-                  - qflx_evap_tot(c)         &
-                  - qflx_surf(c)             &
-                  - qflx_qrgwl(c)            &
-                  - qflx_drain(c)            &
-                  - qflx_drain_perched(c)    &
-                  - qflx_ice_runoff_snwcp(c) &
-                  - qflx_ice_runoff_xs(c)    &
-                  - qflx_snwcp_discarded_liq(c) &
-                  - qflx_snwcp_discarded_ice(c)) * dtime
+                  + qflx_flood_col(c)        &
+                  + qflx_sfc_irrig_col(c)    &
+                  + qflx_glcice_dyn_water_flux_col(c) &
+                  - qflx_evap_tot_col(c)     &
+                  - qflx_surf_col(c)         &
+                  - qflx_qrgwl_col(c)        &
+                  - qflx_drain_col(c)        &
+                  - qflx_drain_perched_col(c) &
+                  - qflx_ice_runoff_col(c)   &
+                  - qflx_snwcp_discarded_liq_col(c) &
+                  - qflx_snwcp_discarded_ice_col(c)) * dtime
 
           else
 
-             errh2o(c) = 0.0_r8
+             errh2o_col(c) = 0.0_r8
 
           end if
 
        end do
        
-       errh2o_max_val = maxval(abs(errh2o(bounds%begc:bounds%endc)))
+       errh2o_max_val = maxval(abs(errh2o_col(bounds%begc:bounds%endc)))
 
        if (errh2o_max_val > h2o_warning_thresh) then
 
-           indexc = maxloc( abs(errh2o(bounds%begc:bounds%endc)), 1 ) + bounds%begc -1
-           write(iulog,*)'WARNING:  water balance error ',&
+           indexc = maxloc( abs(errh2o_col(bounds%begc:bounds%endc)), 1 ) + bounds%begc - 1
+           write(iulog,*)'WARNING:  column-level water balance error ',&
              ' nstep= ',nstep, &
              ' local indexc= ',indexc,&
            ! ' global indexc= ',GetGlobalIndex(decomp_index=indexc, clmlevel=namec), &
-             ' errh2o= ',errh2o(indexc)
+             ' errh2o= ',errh2o_col(indexc)
          if ((errh2o_max_val > error_thresh) .and. (DAnstep > skip_steps)) then
               
               write(iulog,*)'clm urban model is stopping - error is greater than 1e-5 (mm)'
```

Comment:
- [ ] @slevisconsulting this write statement came up in today's ctsm software meeting. I know this predates this PR, but while you're working with this module, could you please:
  - In this line, get rid of 'urban' (that was accidentally left in when this subroutine was refactored a couple of years ago)
  - In many places in this module, we have the text, 'clm model is stopping'; can you please change that to 'CTSM is stopping'?
  - Also, in a number of places where we write this statement about stopping, it says, 'error is greater than 1e-5' or something like that. But actually, the threshold is a parameter, so can you change these write statements to write out the value of the parameter rather than using a hard-coded value that may disagree with the parameter?

Thanks in advance for doing this cleanup of some pre-existing issues with this module.

Reply:
Happy to do it. Cleaning up feels good :-)